libs/api/domains/national-registry-v3/src/lib/nationalRegistryV3.resolver.ts

```diff
@@ -0,0 +1,165 @@
+import { ChildGuardianshipInput } from './graphql/dto/nationalRegistryChildGuardianshipInput'
+import { Spouse } from './graphql/models/nationalRegistrySpouse.model'
+import { Address } from './graphql/models/nationalRegistryAddress.model'
+import { Birthplace } from './graphql/models/nationalRegistryBirthplace.model'
+import { Residence } from './graphql/models/nationalRegistryResidence.model'
+import { Citizenship } from './graphql/models/nationalRegistryCitizenship.model'
+import { Name } from './graphql/models/nationalRegistryName.model'
+import { Religion } from './graphql/models/nationalRegistryReligion.model'
+import { NationalRegistryV3Service } from './nationalRegistryV3.service'
+import { DomicilePopulace } from './graphql/models/nationalRegistryDomicilePopulace.model'
+
+@UseGuards(IdsAuthGuard, IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.meDetails)
+@Resolver(() => Person)
+@Audit({ namespace: '@island.is/api/national-registry-v3' })
+export class NationalRegistryV3Resolver {
+  constructor(private nationalRegistryV3Service: NationalRegistryV3Service) {}
+
+  @Query(() => Person, {
+    name: 'nationalRegistryUserV3',
+    nullable: true,
+  })
+  @Audit()
+  async nationalRegistryPersons(
+    @CurrentUser() user: User,
+  ): Promise<Person | null> {
+    return this.nationalRegistryV3Service.getNationalRegistryPerson(
+      user.nationalId,
+    )
+  }
+
+  @Query(() => ChildGuardianship, {
+    name: 'nationalRegistryUserV3ChildGuardianship',
+    nullable: true,
+  })
+  @Audit()
+  async childGuardianship(
+    @Context('req') { user }: { user: User },
+    @Args('input') input: ChildGuardianshipInput,
+  ): Promise<ChildGuardianship | null> {
+    return this.nationalRegistryV3Service.getChildGuardianship(
+      user,
+      input.childNationalId,
+    )
+  }
+
+  @ResolveField('children', () => [Person], {
+    nullable: true,
+  })
+  @Audit()
+  async resolveChildren(
+    @Context('req') { user }: { user: User },
+    @Parent() person: Person,
+  ): Promise<Array<Person> | null> {
+    if (person.nationalId !== user.nationalId) {
+      return null
+    }
+    return this.nationalRegistryV3Service.getChildren(user.nationalId)
+  }
+
+  @ResolveField('parents', () => [Person], {
+    nullable: true,
+  })
+  @Audit()
+  async resolveParents(
+    @Context('req') { user }: { user: User },
+    @Parent() person: Person,
+  ): Promise<Array<Person> | null> {
+    if (person.nationalId !== user.nationalId) {
+      return null
+    }
+    return this.nationalRegistryV3Service.getParents(user.nationalId)
+  }
+
+  @ResolveField('residenceHistory', () => [Residence], {
+    nullable: true,
+  })
+  @Audit()
+  async resolveResidenceHistory(
+    @Context('req') { user }: { user: User },
+    @Parent() person: Person,
+  ): Promise<Array<NationalRegistryV3Service> | null> {
+    /*return this.nationalRegistryV3Service.getNationalRegistryResidenceHistory(
+      person.nationalId,
+    )*/
```

Comment:
Not ready?

Reply:
Not available in the new service. They only provide a user's current residence, the last icelandic residence a user had, and the residence the user had as of last 1st of december.